### ISSUE TITLE
Complie errors...?

### ISSUE THREAD

**Issue by 0o0TheDude0o0** [MatthewBooth/JFLTE-GPE#22]
This normal activity?

thedude@thedude-VirtualBox:~/android/JFLTE-GPE/ROM$ ./build
Select to either BUILD the ROM, or just the modified apps/jars
1) Build ROM, Kernel and Files	4) Build Modded Files
2) Build ROM			5) Build Addons
3) Build Kernel			6) Exit
#? 1
arch/x86/Makefile:136: CONFIG_X86_X32 enabled but no binutils support
Makefile:652: Cannot use CONFIG_CC_STACKPROTECTOR_REGULAR: -fstack-protector not supported by compiler
arch/x86/Makefile:136: CONFIG_X86_X32 enabled but no binutils support
Makefile:652: Cannot use CONFIG_CC_STACKPROTECTOR_REGULAR: -fstack-protector not supported by compiler

**Comment by 0o0TheDude0o0** [comment]
thedude@thedude-VirtualBox:~/android/JFLTE-GPE/ROM$ sudo dpkg -s binutils
[sudo] password for thedude: 
Package: binutils
Status: install ok installed
Priority: optional
Section: devel
Installed-Size: 11468
Maintainer: Ubuntu Core developers <ubuntu-devel-discuss@lists.ubuntu.com>
Architecture: amd64
Version: 2.24-5ubuntu13
Replaces: binutils-gold (<< 2.20.51.20100415), binutils-mingw-w64-i686 (<< 2.23.52.20130612-1+3), binutils-mingw-w64-x86-64 (<< 2.23.52.20130612-1+3)
Provides: binutils-gold, elf-binutils
Depends: libc6 (>= 2.14), zlib1g (>= 1:1.2.0)
Suggests: binutils-doc (>= 2.24-5ubuntu13)
Breaks: binutils-mingw-w64-i686 (<< 2.23.52.20130612-1+3), binutils-mingw-w64-x86-64 (<< 2.23.52.20130612-1+3)
Conflicts: binutils-gold (<< 2.20.51.20100415), binutils-mingw-w64-i686 (<< 2.23.52.20130612-1+3), binutils-mingw-w64-x86-64 (<< 2.23.52.20130612-1+3), elf-binutils, gas, modutils (<< 2.4.19-1)
Description: GNU assembler, linker and binary utilities
 The programs in this package are used to assemble, link and manipulate
 binary and object files.  They may be used in conjunction with a compiler
 and various libraries to build programs.
Original-Maintainer: Matthias Klose <doko@debian.org>
thedude@thedude-VirtualBox:~/android/JFLTE-GPE/ROM$

**Comment by MatthewBooth** [comment]
Can you try adding [these](http://wiki.cyanogenmod.org/w/Build_for_hammerhead#Install_the_Build_Packages) packages too:

```bash
bison build-essential curl flex git gnupg gperf libesd0-dev liblz4-tool libncurses5-dev libsdl1.2-dev libwxgtk2.8-dev libxml2 libxml2-utils lzop pngcrush schedtool squashfs-tools xsltproc zip zlib1g-dev g++-multilib gcc-multilib lib32ncurses5-dev lib32readline-gplv2-dev lib32z1-dev
```

Some will be doubled from before but that's ok, I'll filter them out later if they work.

**Comment by 0o0TheDude0o0** [comment]
Installed, and started fresh so I'm re-cloning the repo the update way ;) I'll report back.

**Comment by MatthewBooth** [comment]
I'm trying it on a VPS too, so I can get that "just installed" affect. I figured there's some stuff already installed on my machine from AOSP and CM building that is helping.

**Comment by 0o0TheDude0o0** [comment]
I am sure that is the case :)

**Comment by MatthewBooth** [comment]
Dunno how far it got before you got that error, but it's building now for me.

**Comment by 0o0TheDude0o0** [comment]
right off the bat. First thing

**Comment by MatthewBooth** [comment]
I'll let this finish, and then commit the change if all is well

**Comment by 0o0TheDude0o0** [comment]
cool, still waiting on these to clone

**Comment by MatthewBooth** [comment]
I used a DigitalOcean VPS, so it had a 20MB/s connection haha

**Comment by 0o0TheDude0o0** [comment]
Well that must be nice!

**Comment by 0o0TheDude0o0** [comment]
Damn, as soon as I ./build... It continues but I sure don't like those errors...

thedude@thedude-VirtualBox:~/android/JFLTE-GPE/ROM$ ./build
Select to either BUILD the ROM, or just the modified apps/jars
1) Build ROM, Kernel and Files	4) Build Modded Files
2) Build ROM			5) Build Addons
3) Build Kernel			6) Exit
#? 1
arch/x86/Makefile:136: CONFIG_X86_X32 enabled but no binutils support
Makefile:652: Cannot use CONFIG_CC_STACKPROTECTOR_REGULAR: -fstack-protector not supported by compiler
arch/x86/Makefile:136: CONFIG_X86_X32 enabled but no binutils support
Makefile:652: Cannot use CONFIG_CC_STACKPROTECTOR_REGULAR: -fstack-protector not supported by compiler
Building JFLTE-GPE stock kernel

**Comment by 0o0TheDude0o0** [comment]
I am running this on a x64 vm... You doing the same>?

**Comment by 0o0TheDude0o0** [comment]
Kernel/arch/x86/Makefile:136 line 136-146


archscripts:
	$(Q)$(MAKE) $(build)=arch/x86/tools relocs

###
# Syscall table generation

archheaders:
	$(Q)$(MAKE) $(build)=arch/x86/syscalls all

###

**Comment by MatthewBooth** [comment]
Yeah this is a 64 bit machine im using for my PC and the VPS.

Not sure why it's calling the x86 architecture :/

It just finished here on the VPS

**Comment by 0o0TheDude0o0** [comment]
This just fixed it for me!

Kernel/arch/x86/Makefile
Line 138 should read $(Q)$(MAKE) $(build)=arch/x86/toolsrelocs (not $(Q)$(MAKE) $(build)=arch/x86/tools relocs)

Line 144 should read $(Q)$(MAKE) $(build)=arch/x86/syscallsall (not $(Q)$(MAKE) $(build)=arch/x86/syscalls all)

**Comment by MatthewBooth** [comment]
That file shouldn't even be called, though, since we're making for arm, not x86

**Comment by 0o0TheDude0o0** [comment]
I would agree with that.

**Comment by MatthewBooth** [comment]
Not sure about this, but it worked fine on my VPS (though I found an error in the build_kernel script that's unrelated). I guess it's something to do with the VM :(

**Comment by MatthewBooth** [comment]
Open a new issue for this. I'll close this as the original issue is fixed
